bodo/io/iceberg/catalog/snowflake.py

```diff
@@ -0,0 +1,315 @@
+from __future__ import annotations
```

Comment:
This is vendored from a PR from PyIceberg. Sadly, the PR adding this to PyIceberg was closed and abandoned. We should probably take it over.

This is an exact copy right now, but the later PR adding prefetch support and write will extend it.

TODO: Add docstring

Reply:
Added